### ISSUE TITLE
Add time-picker

### ISSUE THREAD

**Issue by brunocoelho** [airbnb/react-dates#255]
Hi guys, any possibility to also support time-picker in the future?

**Comment by oyeanuj** [comment]
@ljharb Just curious about your thoughts or plans on this request?

**Comment by oyeanuj** [comment]
@ljharb Just curious about your thoughts or plans on this request?

**Comment by ljharb** [comment]
To be honest, this seems like a useful thing to be an entirely separate package; which if we wanted a `DateTimePicker`, we'd pull in as a dependency. `react-dates` is already pretty big.

**Comment by mateioprea** [comment]
Hello guys, I did this today: https://www.dropbox.com/s/6z0wu6papx6ff5m/Screenshot%202017-05-22%2018.58.15.png?dl=0 

I'm still working on it, but will submit a pull requests when I finish :)

**Comment by oyeanuj** [comment]
@mateioprea Good job on making it feel part of the plugin, I'd love for this to be an option. One thing you might be missing is an AM/PM meridiem toggle?

@ljharb I hear you on the size, but I wonder if the more effective solution there is to be able to import only the component you need rather than importing the whole package with all the components in them? (for example, even today, when I am using `SingleDatePicker`, it ends up importing `DateRangePicker` in the final build)

**Comment by mateioprea** [comment]
@oyeanuj now that you're saying will work on implementing that too. I thought that 24h format will be fine, but I think AM/PM thing can be optional and it can default to 24h

**Comment by ljharb** [comment]
@oyeanuj that's easily resolved by deep-linking to the thing you want, instead of importing from the "main" entry point - but bundle size isn't what i was talking about.

**Comment by BamaBoy** [comment]
Any workaround on this?

**Comment by smeijer** [comment]
@mateioprea did you ever release that package?

**Comment by mateioprea** [comment]
@smeijer check on my profile. But i used something else and didn’t get a chance to release anything. From what I remember i still had some things to do but it’s a good start

**Comment by oyeanuj** [comment]
@mateioprea nice, hopefully it will end up in a PR soon 😄

**Comment by mateioprea** [comment]
@oyeanun they said that they don’t want that. Maybe act as an independent package

**Comment by fengshuo** [comment]
this might be an alternative option [react-flatpickr](https://github.com/coderhaoxin/react-flatpickr), but

**Comment by email2vimalraj** [comment]
@mateioprea : Release it as an independent package

**Comment by JarLowrey** [comment]
You all might be interested in [react-datetime](https://github.com/YouCanBookMe/react-datetime)

**Comment by Hassan-Zahid** [comment]
Do we have any update on this?

**Comment by sahilchopra** [comment]
+1

**Comment by lionskape** [comment]
Did anything happens last 1.5 year?

**Comment by nathantruant** [comment]
Also still hoping to see something on this.

**Comment by Shumaister** [comment]
we dream with this feature

**Comment by fmaymone** [comment]
would love to see this

**Comment by Sefton419** [comment]
I don't know if this has been mentioned yet, but it would be super cool if they added support for a time-picker ;)

**Comment by jeffj** [comment]
My product (see attached screen shot) requires a time picker. If anyone is working on this and has a branch I would love to lend a hand.

![download](https://user-images.githubusercontent.com/237260/68519892-ca1cb000-0248-11ea-8ee6-7d53553cb3dd.jpg)

**Comment by m-nathani** [comment]
still no support for timepicker ? :laughing:  :hourglass:

**Comment by dhanraj-khatal** [comment]
when we will get time picker support over here ?

**Comment by mehrdadg11** [comment]
Did anything happens last 3 year about timepicker?

**Comment by Xapgkop4uk** [comment]
Unfortunately, react-dates doesn't work with time. I created a wrapper over DateTimePicker to fix it but faced hardcoded hours in sources. https://github.com/airbnb/react-dates/pull/2072/files there required changes. Please take a look. I will appreciate it

**Comment by caub** [comment]
It's pretty useful to be able to select a date and a time/timeslot from one modal interaction

**Comment by cgarrovillo** [comment]
Any more support on this?

**Comment by plamenh** [comment]
Is this dead?

**Comment by thedrivay** [comment]
Nothing happens since 2017🤣

**Comment by ljharb** [comment]
@thedrivay yes, when nobody pays any money or spends any time to contribute, things often don’t happen. Do you think the passage of time will change anything?
